lib/config/validation.ts

```diff
@@ -673,6 +677,29 @@ export async function validateConfig(
         }
       }
     }
+
+    if (key === 'hostRules' && is.array(val)) {
+      const allowedHeader = GlobalConfig.get('allowedHeader');
+      for (const rule of val as HostRule[]) {
+        if (!rule.header) {
+          continue;
+        }
+        for (const [header, value] of Object.entries(rule.header)) {
+          if (!is.string(value)) {
+            errors.push({
+              topic: 'Configuration Error',
+              message: `Invalid hostRules header value configuration: should be a string.`,
+            });
+          }
+          if (!anyMatchRegexOrMinimatch(allowedHeader, header)) {
+            errors.push({
+              topic: 'Configuration Error',
+              message: `hostRules header \`${header}\` is not permitted by this bot's \`allowedHeader\`.`,
```

Comment:
```suggestion
              message: `hostRules header \`${header}\` is not allowed by this bot's \`allowedHeader\`.`,
```

I think we should use allowed, as that matches the terms in the `allowedHeader` config option name.

I think allowed is easier to understand than permitted.